builder/vsphere/common/step_http_ip_discover.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/hashicorp/packer/helper/multistep"
+)
+
+// Step to discover the http ip
+// which guests use to reach the vm host
+// To make sure the IP is set before boot command and http server steps
+type StepHTTPIPDiscover struct {
+	HTTPIP string
+}
+
+func (s *StepHTTPIPDiscover) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ip, err := getHostIP(s.HTTPIP)
+	if err != nil {
+		state.Put("error", err)
+		return multistep.ActionHalt
+	}
+	state.Put("http_ip", ip)
+
+	return multistep.ActionContinue
+}
+
+func (s *StepHTTPIPDiscover) Cleanup(state multistep.StateBag) {}
+
+func getHostIP(s string) (string, error) {
+	if s != "" {
+		if net.ParseIP(s) != nil {
+			return s, nil
+		} else {
+			return "", fmt.Errorf("invalid IP address")
+		}
+	}
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return "", err
+	}
+
+	for _, a := range addrs {
+		ipnet, ok := a.(*net.IPNet)
+		if ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
```

Comment:
Would it make sense to default to the IP that is in the range of the `ip_wait_address`?

Reply:
I think that's a question for a different PR -- this isn't new code, just being moved into its own step. 

I'm not sure the little HTTP server Packer launches even supports IPV6 😬

Reply:
It was just something that I thought about when implementing the `ip_wait_address`. The http server is just the Go http.Server{} which means that it would support ipv6. It would also help with getting the local IP that matches the remote network when multiple NICs are involved.

Reply:
This does make sense to implement in a new PR.